tls/ttls/net.c

```diff
@@ -529,49 +188,45 @@ int mbedtls_net_recv_timeout( void *ctx, unsigned char *buf, size_t len,
  */
 int mbedtls_net_send( void *ctx, const unsigned char *buf, size_t len )
 {
-    int ret;
-    int fd = ((mbedtls_net_context *) ctx)->fd;
-
-    if( fd < 0 )
-        return( MBEDTLS_ERR_NET_INVALID_CONTEXT );
-
-    ret = (int) write( fd, buf, len );
-
-    if( ret < 0 )
-    {
-        if( net_would_block( ctx ) != 0 )
-            return( MBEDTLS_ERR_SSL_WANT_WRITE );
+    struct socket *sock = ((mbedtls_net_context *)ctx)->socket;
+    struct msghdr msg;
+    struct iovec iov;
+    int size;
+    mm_segment_t oldfs;
+
+    if(sock == NULL) {
+        return -1;
+    }
```

Comment:
CodingStyle:

Do not unnecessarily use braces where a single statement will do.              
        
        if (condition)
                action();